docker/archive/archive_transport.go

```diff
@@ -0,0 +1,110 @@
+package archive
+
+import (
+	"os"
+
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/types"
+)
+
+// Transport is an ImageTransport for Docker images stored on the local filesystem (as standalone archives).
+var Transport = archiveTransport{}
+
+type archiveTransport struct{}
+
+// Name returns the name of the transport, which must be unique among other transports.
+func (t archiveTransport) Name() string {
+	return "docker-archive"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an ImageReference.
+func (t archiveTransport) ParseReference(reference string) (types.ImageReference, error) {
+	return ParseReference(reference)
+}
+
+// ValidatePolicyConfigurationScope checks that scope is a valid name for a signature.PolicyTransportScopes keys
+// (i.e. a valid PolicyConfigurationIdentity() or PolicyConfigurationNamespaces() return value).
+// It is acceptable to allow an invalid value which will never be matched, it can "only" cause user confusion.
+// scope passed to this function will not be "", that value is always allowed.
+func (t archiveTransport) ValidatePolicyConfigurationScope(scope string) error {
+	// FIXME FIXME
+	return nil
+}
+
+// archiveReference is an ImageReference for Docker images stored on the local filesystem.
+type archiveReference string // FIXME FIXME
```

Comment:
Overall, the `docker-daemon:` `ImageReference` implementation is unfinished, and not a good model ATM. We will be fixing that soonish ( #143 ).

If the semantics of the `docker-archive:` references is that they are simply paths to tarballs, probably see what the `dir:` transport does for `ValidatePolicyConfigurationScope`, `StringWithinTransport`, `PolicyConfiguration{Identity,Namespaces`} etc.

Note that because these things interact with signature validation, the `_transport.go` files really should have tests covering every single line if possible.